[WIP] SplitMongo Pruning

eltoncarr-ms opened the pull request:
> **This is work in progress**

The target of this PR is to complete initial review of the script for handling the pruning of the mongo database modulestore structures.

Core capabilities currently available:
1. querying specified edxapp database active_versions and structures collections 
2. identifying all structures that can be removed
3. removing identified structures

Additional capabilities that are in progress (or targeted):
1. addressing performance concerns about pulling all documents in active_versions and structures collections at once
2. enabling use of static test data for easier validation & plugging into the **tox** test framework that tubular uses
3. enabling visualization of before/after outcomes of the pruning action

For this round of reviews, the focus areas are:
1. matching the style and essence of the edx/tubular repository
2. confirming correctness of the pruning activity

eltoncarr-ms committed:
Initial Commit

	modified:   .gitignore
	modified:   requirements.txt
	new file:   scripts/prune_modulestore.py
	modified:   setup.cfg

eltoncarr-ms committed:
Improving messaging & fixing list reference for module structures to remove

	modified:   scripts/prune_modulestore.py

eltoncarr-ms committed:
Adding test support

	modified:   scripts/prune_modulestore.py
	new file:   tubular/tests/prune_test_dataset/dataset.json

eltoncarr-ms committed:
Adding option for removing the original version

	modified:   scripts/prune_modulestore.py

eltoncarr-ms committed:
Added capability to prune static test data

eltoncarr-ms committed:
Linting changes

eltoncarr-ms committed:
Moving prune script to appropriate location

	typechange: prune_modulestore.py
	copied:     prune_modulestore.py -> ../tubular/scripts/prune_modulestore.py

eltoncarr-ms committed:
Refactoring

	modified:   tubular/scripts/prune_modulestore.py

Commit:
Adding code for relinking

Commit:
fixing indentation

eltoncarr-ms committed:
Adding unit test stub

	new file:   tubular/tests/test_prune_modulestore.py
	renamed:    tubular/tests/prune_test_dataset/dataset.json -> tubular/tests/test_prune_modulestore_data.json

eltoncarr-ms committed:
Merge pull request #2 from manikarthikk/mani_relink

[WIP]

Commit:
correcting  the  typo

btelnes commented on tubular/tests/test_prune_modulestore.py:
```diff
@@ -0,0 +1,27 @@
+        self.assertEqual((True, True)
+    
+
+if __name__ == '__main__':
```
Travis CI build fails at line 25 with "SyntaxError: invalid syntax"

eltoncarr-ms replied:
Fixed. line 22 did not have a closing parenthesis

eltoncarr-ms committed:
Merge pull request #3 from manikarthikk/patch-1

[WIP]: Correcting the Typo

eltoncarr-ms committed:
Refactoring

	modified:   tubular/scripts/prune_modulestore.py

eltoncarr-ms committed:
WIP - Enabling test case & fixing error with main

	modified:   tubular/tests/test_prune_modulestore.py

eltoncarr-ms committed:
Lint fixes

	modified:   tubular/scripts/prune_modulestore.py
	modified:   tubular/tests/test_prune_modulestore.py

eltoncarr-ms committed:
Linting updates

	modified:   tubular/scripts/prune_modulestore.py
	modified:   tubular/tests/test_prune_modulestore.py

eltoncarr-ms committed:
Linting updates

eltoncarr-ms committed:
Linting updates

	modified:   tubular/scripts/prune_modulestore.py
	modified:   tubular/tests/test_prune_modulestore.py

eltoncarr-ms committed:
Enabling return value for main script

	modified:   tubular/scripts/prune_modulestore.py

eltoncarr-ms committed:
Removed explicit variable for deleting original version

	modified:   tubular/scripts/prune_modulestore.py
	modified:   tubular/tests/test_prune_modulestore.py

eltoncarr-ms committed:
Updating pymongo requirement

	modified:   requirements.txt

eltoncarr-ms committed:
Enabling relink for static dataset validation

	modified:   tubular/scripts/prune_modulestore.py

eltoncarr-ms committed:
Linting updates

	modified:   tubular/scripts/prune_modulestore.py

eltoncarr-ms committed:
Refactoring

	modified:   tubular/scripts/prune_modulestore.py
	modified:   tubular/tests/test_prune_modulestore_data.json

eltoncarr-ms committed:
Refactor - update git ignore items

	modified:   .gitignore

eltoncarr commented:
@btelnes  & @manikarthikk , thanks. I will wrap up this initial iteration
